gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseKey.java

```diff
@@ -172,20 +172,21 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  protected DatastoreV1.Key toPb() {
-    DatastoreV1.Key.Builder keyPb = DatastoreV1.Key.newBuilder();
-    DatastoreV1.PartitionId.Builder partitionIdPb = DatastoreV1.PartitionId.newBuilder();
-    if (projectId != null) {
-      partitionIdPb.setDatasetId(projectId);
+  protected com.google.datastore.v1beta3.Key toPb() {
+    com.google.datastore.v1beta3.Key.Builder keyPb = com.google.datastore.v1beta3.Key.newBuilder();
+    com.google.datastore.v1beta3.PartitionId.Builder partitionIdPb = 
+        com.google.datastore.v1beta3.PartitionId.newBuilder();
+    if (!Strings.isNullOrEmpty(projectId)) {
```

Comment:
Actually, I now wonder if we could also just change projectId and namespace to default to an empty string and always provide paristionId. We need to check with the Datastore that if partition Id is provided with an empty project_id will they consider it as "no-partition-id-provided-use-default-instead".